### PR TITLE
helm: Add CA Pinning Support 

### LINKED ISSUE #13580: Add ability to set ca_pin in Helm chart
Provide ability to set ca_pin in kube-agent helm chart.

See https://github.com/gravitational/teleport/issues/12444 for more context.

### PR DESCRIPTION
**Pull request by hugoShaka**
Fixes #13580 by adding an additional `caPin` field allowing the user to set the `ca_pin` config field.

### CODE REVIEW

**Comment by marcoandredinis** [comment]
I think the caPin can have multiple CAs :thinking: 
Can you please confirm?
If so, it should either be a list of strings or we should mention the split char

If I'm not mistaken, this was the PR that add support for it
https://github.com/gravitational/teleport/pull/12796

**Comment by marcoandredinis** [comment]
Other than that, LGTM

**Review by marcoandredinis — APPROVED** [review]

File: docs/pages/kubernetes-access/helm/reference/teleport-kube-agent.mdx [hunk 1]
```diff
@@ -357,6 +357,34 @@ See [this link for information on Community Docker image versions](../../../setu
   </TabItem>
 </Tabs>
 
+## `caPin`
+
+| Type | Default value |
+| - | - |
+| `list` | `[]` |
+
+When `caPin` is set, the teleport agent will use its values to check the
```
Comment (webvictim) [inline docs/pages/kubernetes-access/helm/reference/teleport-kube-agent.mdx:366]:
```suggestion
When `caPin` is set, the Teleport agent will use its values to check the
```

File: docs/pages/kubernetes-access/helm/reference/teleport-kube-agent.mdx [hunk 2]
```diff
@@ -357,6 +357,34 @@ See [this link for information on Community Docker image versions](../../../setu
...
+Auth server identity during the initial join. This enables a more secure
```
Comment (webvictim) [inline docs/pages/kubernetes-access/helm/reference/teleport-kube-agent.mdx:367]:
```suggestion
identity of the Auth server during the initial join. This enables a more secure
```

**Review by webvictim — APPROVED** [review]
LGTM with a couple of docs nits.

I'm not actually 100% sure whether we check the CA pin when joining via proxy? It's only usually used for joining via auth.

**Comment by ptgott** [comment]
Hi @hugoShaka, would you mind rebasing so this change applies to the newly moved `teleport-kube-agent` reference? I'll give this a look then.

**Comment by hugoShaka** [comment]
> Hi @hugoShaka, would you mind rebasing so this change applies to the newly moved `teleport-kube-agent` reference? I'll give this a look then.

Done

File: docs/pages/setup/helm-reference/teleport-kube-agent.mdx [hunk 3]
```diff
@@ -357,6 +357,34 @@ See [this link for information on Community Docker image versions](../../setup/g
   </TabItem>
 </Tabs>
 
+## `caPin`
+
+| Type | Default value |
+| - | - |
+| `list` | `[]` |
+
+When `caPin` is set, the teleport agent will use its values to check the
```
Comment (ptgott) [inline docs/pages/setup/helm-reference/teleport-kube-agent.mdx:366]:
```suggestion
When `caPin` is set, the Teleport pod will use its values to check the
```

File: docs/pages/setup/helm-reference/teleport-kube-agent.mdx [hunk 4]
```diff
@@ -357,6 +357,34 @@ See [this link for information on Community Docker image versions](../../setup/g
...
+Auth server identity during the initial join. This enables a more secure
```
Comment (ptgott) [inline docs/pages/setup/helm-reference/teleport-kube-agent.mdx:367]:
```suggestion
Auth Service's identity when first joining a cluster. This enables a more secure
```

File: docs/pages/setup/helm-reference/teleport-kube-agent.mdx [hunk 5]
```diff
@@ -357,6 +357,34 @@ See [this link for information on Community Docker image versions](../../setup/g
...
+way of adding new nodes to a cluster. See
```
Comment (ptgott) [inline docs/pages/setup/helm-reference/teleport-kube-agent.mdx:368]:
```suggestion
way of adding new Teleport instances to a cluster. See
```

File: docs/pages/setup/helm-reference/teleport-kube-agent.mdx [hunk 6]
```diff
@@ -357,6 +357,34 @@ See [this link for information on Community Docker image versions](../../setup/g
...
+["Adding Nodes to the Cluster"](../admin/adding-nodes.mdx).
+
+Each list element can be the pin itself (recommended, works out of the box),
+or a path to a file containing the pin. For the latter it is your
+responsibility to mount the file, using [extraVolumes](#extraVolumes).
```
Comment (ptgott) [inline docs/pages/setup/helm-reference/teleport-kube-agent.mdx:373]:
```suggestion
responsibility to mount the file using [`extraVolumes`](#extraVolumes).
```

**Review by ptgott — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Pull request merged**